src/writer/proposal.ts

```diff
@@ -46,6 +46,11 @@ export async function verify(body): Promise<any> {
     return Promise.reject('wrong proposal format');
   }
 
+  const tsInt = (Date.now() / 1e3).toFixed();
+  if (msg.payload.end < tsInt) {
```

Comment:
I feel like we can make it "Not to be in past"
```suggestion
  if (msg.payload.end <= tsInt) {
```